### PR TITLE
SALTO-2933 SALTO-2497 SALTO-2268 SF add messages to the error retry list 

### PR DESCRIPTION
**Pull request by Uri-Bar**
_add messages to the error retry list _

---
_Additional context for reviewer_:
None

---
_Release Notes_: 
Salesforce-adapter:

* Fetch stability improvements
---

_User Notifications_: 
None

### CODE REVIEW

**Review by amitgoldarb — APPROVED** [review]
This looks great.
I think the release notes here are a bit incorrect though. The release notes here are meant for the clients of salto. I think that this is TMI, I would write this how it would improve the user experience when using salto - "Fetch stability improvements"

File: packages/salesforce-adapter/src/client/client.ts [hunk 1]
```diff
@@ -474,11 +478,13 @@ export default class SalesforceClient {
   }
 
   private retryOnBadResponse<T extends object>(request: () => Promise<T>): Promise<T> {
+    const retryAttempts = this.retryOptions.maxAttempts ?? DEFAULT_RETRY_OPTS.maxAttempts
     const requestWithRetry = async (attempts: number): Promise<T> => {
       let res: T
       try {
         res = await request()
       } catch (e) {
+        log.warn(`caught exception: ${e.message}. ${attempts} retry attempts left from ${retryAttempts} in total`)
```
Comment (adamro) [inline packages/salesforce-adapter/src/client/client.ts:487]: What's the idea behind having both this and the log in line 489?

**Review by adamro — COMMENTED** [review]
Is this in sync with https://github.com/salto-io/salto/pull/3595? How did we end up having 2 PRs so similar? :D

File: packages/salesforce-adapter/src/client/client.ts (hunk 1 above)
Comment (Uri-Bar) [reply packages/salesforce-adapter/src/client/client.ts:487]: the retry mechanism does not work as expected.. we see some error message that are in the list but did not preform any retry so we want to be able to debug it better

**Pull request merged**